### PR TITLE
Building your first telegram bot in Node.js

### PR DESCRIPTION
**Pull request by calebroHQ**
In this tutorial, you will learn how to build telegram bots using node.js runtime environment and telegraf library from scratch. This will act as an introduction to the Telegram bot environment that will enable you to build weather bot, vendor bots, e-commerce bot, or even dictionary bot. This tutorial will be as beginner-friendly as possible.

image1, image2,image3 are screenshots from the telegram bot I just created.

### CODE REVIEW

**Comment by cyberShaw** [comment]
Hey, @calebroHQ!
Could you please replace the hero image with another that is of dimensions 600x400?

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+
+# Building your first Telegram bot in Node.js and Telegraf library 
+
+## Introduction
+
+What are telegram bots? Bots are third-party application accounts that run inside the Telegram application. Users can interact with bots by sending them messages, commands, and inline requests. You control your bots using HTTPS requests from the Telegram Bot API.
+
+Telegram bots have made it easy to automate task in telegram channels and groups. You can even build your own telegram bot to curate news or monitor cryptocurrencies trend for you.
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:8]:
```suggestion
Telegram bots have made it easy to automate tasks in telegram channels and groups. You can even build your own telegram bot to curate news or monitor cryptocurrencies trend for you.
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+In this tutorial, you will learn how to build telegram bots using node.js runtime environment and telegraf library from scratch. This will act as an introduction to the Telegram bot environment that will enable you to build weather bot, vendor bots, e-commerce bot, or even dictionary bot. This tutorial will be as beginner-friendly as possible.
+
+## Prerequisite
+Before you begin this guide you’ll need the following:
+
+1. [Node.js](https://nodejs.org/en/download/) installed
+2. npm installed
+3. [Visual Studio Code](https://code.visualstudio.com/download) installed
+4. Basic knowledge on using the CLI
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:18]:
```suggestion
4. Basic knowledge of using the CLI
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+5. A [Telegram](https://telegram.org/) account 
+
+## Setting up your bot
+
+First go to the the telegram search bar and search @botfather (this is a special telegram bot that controls all others).We are going to use this to get a token for our new bot. There are several commands associated with @botfather , we are going to start with /newbot then provide the desired name for the bot. The name must always end with 'bot'. Now that we have the token we can write the code.
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:23]:
```suggestion
First, go to the telegram search bar and search @botfather (this is a special telegram bot that controls all others). We are going to use this to get a token for our new bot. There are several commands associated with @botfather, we are going to start with /newbot then provide the desired name for the bot. The name must always end with 'bot'. Now that we have the token we can write the code.
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+## The coding
+Lets first create a folder for our new project, i.e: myfirstbot. Open the command line and navigate to the folder. Initialize the project by running the following commands to install telegraf:
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:26]:
```suggestion
Let us first create a folder for our new project, i.e: myfirstbot. Open the command line and navigate to the folder. Initialize the project by running the following commands to install telegraf:
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    })
+
+})
+
+bot.launch();
+```
+
+## Using the bot
+
+We have successfully written two blocks of codes, one thats welcomes you when you start the bot and the other sends you images when you click inline keyboard buttons. Run the bot in the terminal and then go to the telegram search bar and search the name of your bot.
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:94]:
```suggestion
We have successfully written two blocks of codes, one that welcomes you when you start the bot and the other sends you images when you click inline keyboard buttons. Run the bot in the terminal and then go to the telegram search bar and search the name of your bot.
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+Here are the results from telegram.
+
+![bot results](image1.jpg)
+![bot results](image2.jpg)
+![bot results](image3.jpg)
+
+## Summary
+
+In this article we have gone through how to get your telegram bot token, setting up your environment and writing a simple bot for displaying imagesof animals using inline keyboard feature.
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:108]:
```suggestion
In this article we have gone through how to get your telegram bot token, setting up your environment and writing a simple bot for displaying images of animals using the inline keyboard feature.
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+We have successfully written two blocks of codes, one thats welcomes you when you start the bot and the other sends you images when you click inline keyboard buttons. Run the bot in the terminal and then go to the telegram search bar and search the name of your bot.
+
+```bash
+node app.js
+```
+
+Here are the results from telegram.
+
+![bot results](image1.jpg)
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:102]:
```suggestion
![bot result 1](/engineering-education/telegram-bot-in-nodejs/image1.jpg)
```

Image paths should be of form, /engineering-education/telegram-bot-in-nodejs/image1.jpg

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+![bot results](image2.jpg)
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:103]:
```suggestion
![bot result 2](/engineering-education/telegram-bot-in-nodejs/image2.jpg)
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+![bot results](image3.jpg)
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:104]:
```suggestion
![bot result 3](/engineering-education/telegram-bot-in-nodejs/image3.jpg)
```

File: articles/Building-your-first-Telegram-bot-in-Node.js/index.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+```javascript
+bot.command('start', ctx => {
+    console.log(ctx.from)
+    bot.telegram.sendMessage(ctx.chat.id, 'hello there! Welcome to my new telegram bot.', {})
+})
+```
+
+To make it more interesting lets write code that shows you images of animals when you click inline keyboard buttons. Create a res folder and add images into it.
+
+```javascript
```
Comment (cyberShaw) [inline articles/Building-your-first-Telegram-bot-in-Node.js/index.md:52]: Please add code comments wherever you write code.
This ensures that beginners can understand the code

**Review by cyberShaw — CHANGES_REQUESTED** [review]
Great article on telegram bots, @calebroHQ!

The article is of only 500+ words. The requirement demands the article to be of 750 words minimum. Please add more content to the article and make the changes that I've suggested below and we can move on to the next round of review.

**Comment by cyberShaw** [comment]
@calebroHQ Please pull changes before making the changes

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by calebroHQ** [comment]
@cyberShaw I have made the revision as suggested.

**Comment by cyberShaw** [comment]
Thank you for making the changes, @calebroHQ!

I have two minor changes that need to be made:

1. Please add a hero image of dimensions 600 x 400. The hero image you added is of 600 x 368
2. Add a GitHub link that contains the final code for this tutorial.

**Comment by calebroHQ** [comment]
@cyberShaw hey, It's all done.

**Review by cyberShaw — APPROVED** [review]

**Comment by hectorkambow** [comment]
great work gentlemen @calebroHQ 

cc @cyberShaw

**Pull request merged**